src/main/java/org/apache/commons/math4/stat/StatUtils.java

```diff
@@ -107,7 +113,7 @@ public static double sum(final double[] values) throws MathIllegalArgumentExcept
      * @throws MathIllegalArgumentException if the array is null or the array index
      *  parameters are not valid
      */
-    public static double sum(final double[] values, final int begin, final int length)
+    public static double sum(final double @MinLen(1) [] values, final @IndexFor("#1") int begin, final @NonNegative @LTLengthOf(value = {"#1"}, offset = {"#2 - 1"}) int length)
```

Comment:
Could the same reasoning as for `evaluate` be applied to methods in this class? (Removing `@MinLen(1)` and making begin `@IndexOrHigh`? It seems that the methods generally return `NaN`, so passing an empty array is not considered to be an error.